### ISSUE TITLE
Bug about input(type="number") 

### ISSUE THREAD

**Issue by showonne** [angular/angular.js#15491]
***Note*: for support questions, please use one of these channels: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#question. This repository's issues are reserved for feature requests and bug reports.**

**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

Input 11111111111111111
Output value is 11111111111111112

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://plnkr.co or similar (template: http://plnkr.co/edit/tpl:yBpEi4).**

http://jsbin.com/bolukerika/edit?html,js,output

**What is the expected behavior?**

The expected output value is 11111111111111111

**What is the motivation / use case for changing the behavior?**

**Which versions of Angular, and which browser / OS are affected by this issue? Did this work in previous versions of Angular? Please also test with the latest stable and snapshot (https://code.angularjs.org/snapshot/) versions.**

Demo's Angular.js version is 1.4.0

**Other information (e.g. stacktraces, related issues, suggestions how to fix)**

I found the [numberFormatterParser](https://github.com/angular/angular.js/blob/master/src/ng/directive/input.js#L1513) method is the result of this bug, but remove it will fail to pass other test cases.

**Comment by gkalpak** [comment]
The problem is not with the `numberFormatterParser` method per se, but with JavaScript's `Number` type. Actually, it is not a problem; it is a known limitation of Floating Point Arithmetic (there are several resources available on this subject).

So, the problem is that there are certain numbers that cannot be represented using the [IEEE 754 standard][1] (which JavaScript is using). The maximum integer that can be accurately represented (and thus is considered safe) is `Number.MAX_SAFE_INTEGER === 9007199254740991`, which is smaller than `11111111111111111`. In fact, if you write `var num = 11111111111111111`, you will find out that the value of `num` is actually `11111111111111112`.

**tl;dr**
There nothing that can be done in Angular (or JavaScript in general) to address this problem. It is a fundamental limitation in the language.

[1]: https://en.wikipedia.org/wiki/IEEE_floating_point

**Comment by showonne** [comment]
@gkalpak Thanks for your answer :D, I have also considered it is a limitation of Floating Point Arithmetic before, just because it works as my expected in another framework, [demo](http://jsbin.com/cinucusani/edit?html,js,output), I opened this issue.

**Comment by gkalpak** [comment]
@showonne, this _seems_ to be so, because in another framework the model value is stored as a string, not a number ([demo][1]). In Angular, a model bound to a `number` input is converted to an actual number.

[1]: http://jsbin.com/zawohutuqe/1/edit?html,js,output
